example/src/main/java/com/ryanharter/auto/value/gson/example/UnrecognizedExample.java

```diff
@@ -0,0 +1,36 @@
+// modified by mapbox
+package com.ryanharter.auto.value.gson.example;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.ryanharter.auto.value.gson.UnrecognizedJsonProperties;
+
+import java.util.Map;
+
+@AutoValue
+public abstract class UnrecognizedExample {
+
+  public abstract int knownField();
+
+  public static UnrecognizedExample.Builder builder() {
+    return new AutoValue_UnrecognizedExample.Builder();
+  }
+
+  public static TypeAdapter<UnrecognizedExample> typeAdapter(Gson gson) {
+    return new AutoValue_UnrecognizedExample.GsonTypeAdapter(gson);
+  }
+
+  @Nullable
+  @UnrecognizedJsonProperties
+  abstract Map<String, Object> unknownProperties();
```

Comment:
```suggestion
  abstract Map<String, JsonElement> unknownProperties();
```

Is that always true? I understand yes, if that's the case we can even enforce it as part of the annotation processor (that the annotated type is always `Map<String, JsonElement>`). That could be the next step though, not a blocker.

Reply:
It would be if we didn't need java serialization. Right now it's always `SerializableWrapper`

Reply:
BTW, is transition later from `Map<String, Object>` to `Map<String, JsonElement>` can be considered as a breaking change? Java's `Map` isn't covariant like Kotlin's one. Scenarios like `object.unknownProperties().put("new key", "new value")` could be broken. I will check if I can make the map immutable under the hood

Reply:
Updated to `abstract Map<String, SerializableJsonElement> unknownProperties();`